### PR TITLE
[MRG+1] MAINT moved _backports to cbook module

### PR DESCRIPTION
**Pull request by NelleV**
This moves the _backports module to the cbook one. 
The cbook.py is now replaced by a folder type module. API is not changed, but it thus means we can start splitting the cbook module into submodules of different categories (deprecation, backports & compatibility, validations etc).

This is just some reorganization of modules to facilitate organization.

### CODE REVIEW

File: lib/mpl_toolkits/mplot3d/art3d.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 from matplotlib import (
     artist, cbook, colors as mcolors, lines, text as mtext, path as mpath)
-from matplotlib._backports import numpy as _backports_np
+from matplotlib.cbook._backports as _backports_np
```
Comment (naoyak) [inline lib/mpl_toolkits/mplot3d/art3d.py]: Should this be:
```python
from matplotlib.cbook._backports import numpy as _backports_np
```

Comment (NelleV) [inline lib/mpl_toolkits/mplot3d/art3d.py]: I definitely messed something up here :)

**Comment by anntzer** [comment]
I would suggest keeping one backport module per external module.  So either `cbook._backports_numpy`, or if want to overengineer it, `cbook._backports.numpy` (anyways it's private :-))

I know right now it's only numpy (and I don't particularly hope that there would be more stuff) but `from .cbook import _backports as _backports_np` just seems wrong if `_backports` is supposedly generic.

Otherwise, +1 to the idea.

**Comment by NelleV** [comment]
I'd hope that we only have numpy backport, but we do also have backports of the subprocess modules in the compat module. The problem is that we need a deprecation cycle to move it to the same place (it would have be nice to have that module private, but it ain't).

Note : @tacaswell it seems that the subprocess module has been renamed in 2.x without deprecation warning. Is that normal? If it is normal, shouldn't we move it to the cbook module to have everything in one place?

**Comment by NelleV** [comment]
(other solution for 2.x, make it private, this way we can do whatever we want with it. But it's really short notice for the release…)

**Comment by tacaswell** [comment]
Please don't over engineer this :smile: 

I am +0.5 on this.  Positive on the extra organization, but concerned of the added indirection going from a single file -> folder and the slight lose in ability to do git forensics (which maybe important for cbook as @anntzer sorts through what of that we are actually using and what of that we can deprecate). 

@NelleV renaming the module was probably a mistake, where did that get done?  It looks like it was renamed in f41000d6f5ca2dc15e2c99da29b501185260d241 which was pre 1.3, but I might be missing something.

**Comment by NelleV** [comment]
mmh, maybe I am wrong. I am looking at git blame, and thus a filename change would not be visible in the blame.

**Comment by NelleV** [comment]
I fixed @anntzer's comment.

**Comment by NelleV** [comment]
We really need to fix our PEP8 compliance tests system… It's raising errors on code I have not modified.

**Comment by QuLogic** [comment]
You've renamed `lib/matplotlib/cbook.py` → `lib/matplotlib/cbook/__init__.py` but forgot to add the new name to the exception list. But fixing the PEP8 errors is better, anyway.

**Comment by NelleV** [comment]
That's exactly my point. We should not have to do this.

**Comment by QuLogic** [comment]
Unfortunately, there's no such thing as a rename in native git, so I'm not sure looking at a diff will work either. I might not be trying it out right, but I wasn't able to get `flake8-diff` to work properly on the changes in this PR.

**Comment by anntzer** [comment]
Thanks!

**Pull request merged**